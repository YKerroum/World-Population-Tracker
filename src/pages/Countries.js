import React from 'react';

const Books = () => (
  <CountriesContainer />
);

export default Books;