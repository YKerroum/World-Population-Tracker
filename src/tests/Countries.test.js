import React from 'react';
import renderer from 'react-test-renderer';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Countries from '../pages/Countries';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn(),
  useLocation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(() => jest.fn()),
}));

describe('Countries component', () => {
  it('should render correctly when loading is true', () => {
    useParams.mockImplementation(() => ({
      continentName: 'Africa',
    }));
    useSelector.mockImplementation(() => ({
      countries: [],
      loading: true,
    }));

    const tree = renderer.create(<Countries />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
