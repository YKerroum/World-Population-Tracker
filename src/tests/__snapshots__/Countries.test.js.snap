// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries component should render correctly when loading is true 1`] = `
<div
  className="countriesCards"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <h1>
    Loading ...
  </h1>
  <div
    className="spinner-grow text-info"
  />
</div>
`;
